packages/schemas/src/gen/utils.ts

```diff
@@ -124,3 +151,23 @@ export const getType = (
     default:
   }
 };
+
+export const parseType = (value: string) => {
+  const typeName = removeParentheses(value);
+  const type = getType(typeName);
+
+  const isString = type === 'string';
+  const parenthesesMatch = findFirstParentheses(value);
+  const maxLength = conditional(
```

Comment:
I am thinking maybe giving a stringMaxLength props might be better. As technically array could have max length as well.

Reply:
I renamed `maxLength` to `stringMaxLength` to keep it clear for now.

The better solution: parse all type-related properties in one place like `parseType` and return encapsulated object (e.g. re-org properties: primitiveType, tsType, isEnum, isArray). But it needs a huge effort and its ROI seems to be low for now, so keep it?